plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1193,6 +1197,11 @@ private void executeDropExtendedStats(ConnectorSession session, IcebergTableExec
 
         Table icebergTable = catalog.loadTable(session, executeHandle.getSchemaTableName());
         beginTransaction(icebergTable);
+        UpdateStatistics updateStatistics = transaction.updateStatistics();
+        for (StatisticsFile statisticsFile : icebergTable.statisticsFiles()) {
+            updateStatistics.removeStatistics(statisticsFile.snapshotId());
```

Comment:
Currently, even after executing:

- `drop_extended_stats` 
- `remove_orphan_files`

the stats file are still to be found in `/data` directory of the table.

When should these files be actually deleted? Can be addressed eventually in a follow-up PR.

Reply:
A no-longer-referenced stats file cannot be distinguished from a not-yet-referenced stats file, so you need to dial down the retention_threshold to have them removed faster.

```
$ find /var/folders/dr/jhnfvsrd2zl925dyfsctj0pr0000gn/T/TrinoTest8535621821089288829/iceberg_data/tpch/nation-f16478597da945cf9b7cc316177222c9 -name \*.stats | tee /dev/stderr | wc -l
0
```

```sql
trino:tpch> ANALYZE nation;
```

```
$ find /var/folders/dr/jhnfvsrd2zl925dyfsctj0pr0000gn/T/TrinoTest8535621821089288829/iceberg_data/tpch/nation-f16478597da945cf9b7cc316177222c9 -name \*.stats | tee /dev/stderr | wc -l
/var/folders/dr/jhnfvsrd2zl925dyfsctj0pr0000gn/T/TrinoTest8535621821089288829/iceberg_data/tpch/nation-f16478597da945cf9b7cc316177222c9/metadata/20221216_114530_00030_sy2sm-db9cd9f6-1554-4df6-a4df-82ef4dc18d32.stats
1
```

```
trino:tpch> ALTER TABLE nation EXECUTE drop_extended_stats();
trino:tpch> SET SESSION iceberg.remove_orphan_files_min_retention = '0s';
trino:tpch> ALTER TABLE nation EXECUTE remove_orphan_files(retention_threshold => '0s');
```

```
$ find /var/folders/dr/jhnfvsrd2zl925dyfsctj0pr0000gn/T/TrinoTest8535621821089288829/iceberg_data/tpch/nation-f16478597da945cf9b7cc316177222c9 -name \*.stats | tee /dev/stderr | wc -l
0
```